Text/PrettyPrint/Compact/Core.hs

```diff
@@ -88,46 +122,78 @@ pareto' acc (x:xs) = if any (≺ x) acc
                             -- for all y ∈ acc, y <= x, and thus x ≺ y
                             -- is false. No need to refilter acc.
 
+leFst :: Ord a => (a, b) -> (a, b) -> Bool
+leFst = (<=) `on` fst
 
```

Comment:
Good point.

If I understand the inner workings right, probably the better change would be

```diff
bests :: forall a. (Ord a, Poset a) => [[a]] -> [a]
bests :: forall a. (Ord a, Poset a) => [[(a, b)]] -> [(a, b)]
```

This will let us remove questionable `Poset (a, b)` instance too, and we won't need `leFst`.